src/main/kotlin/net/djvk/fireflyPlaidConnector2/transactions/PlaidFireflyTransaction.kt

```diff
@@ -0,0 +1,127 @@
+package net.djvk.fireflyPlaidConnector2.transactions
+
+import net.djvk.fireflyPlaidConnector2.api.firefly.models.TransactionTypeProperty
+import net.djvk.fireflyPlaidConnector2.api.plaid.models.Transaction
+import java.time.OffsetDateTime
+import java.time.ZoneId
+
+/**
+ * Represents a single non-transfer transaction, including both the Plaid and the Firefly representation of it.
+ */
+sealed interface PlaidFireflyTransaction {
+    val amount: Double
+    val transactionId: String
+    val fireflyAccountId: Int
+    val fireflyTransaction: FireflyTransactionDto?
+    val plaidTransaction: Transaction?
+    fun getTimestamp(zoneId: ZoneId): OffsetDateTime
+
+    companion object {
+        /**
+         * Wraps Plaid and Firefly transactions into PlaidFireflyTransaction objects, joining them together when
+         * matching transactions are identified.
+         */
+        fun match(
+            plaidTxs: List<Transaction>,
+            fireflyTxs: List<FireflyTransactionDto>,
+            accountMap: Map<String, Int>
+        ): List<PlaidFireflyTransaction> {
+            val plaidById = plaidTxs.groupBy { FireflyTransactionExternalIdIndexer.getExternalId(it.transactionId) }
+            val fireflyByExtId = fireflyTxs.groupBy { it.tx.externalId }
+            val externalIds = plaidById.keys.union(fireflyByExtId.keys)
+
+            return externalIds.flatMap {
+                val matchingPlaid = plaidById[it] ?: listOf()
+                val matchingFirefly = fireflyByExtId[it] ?: listOf()
+
+                // For all transactions that do not have an external ID, or if we've found more matching transactions
+                // than we expected to find, return them without attempting to combine.
+                if (it == null || matchingFirefly.size > 1 || matchingPlaid.size > 1) {
+                    val convertedFirefly = matchingFirefly.map { FireflyTransaction(it) }
+                    val convertedPlaid = matchingPlaid.map {
+                        val accountId = accountMap[it.accountId]
+                        if (accountId == null) {
+                            throw throw IllegalArgumentException("Can not match Plaid transactions from accounts not mapped "
+                                    + "to a Firefly account id")
+                        }
+                        PlaidTransaction(it, accountId)
+                    }
+                    val converted: List<PlaidFireflyTransaction> = convertedPlaid + convertedFirefly
```

Comment:
This type declaration is unnecessary.

Reply:
Fixed in https://github.com/nprzy/firefly-plaid-connector-2/commit/bcb576c2e70402d7fbcb77bd3ee822b51869d7f3